### ISSUE TITLE
Softlock

### ISSUE THREAD

**Issue by mattbruv** [1r3n33/bomberworld#1]
I just experienced a softlock on my first playthrough, it's stuck executing in a loop from `B8DB`-`B8DE`

![](https://i.imgur.com/vHmzcuH.png)

This is the screen it froze on

![](https://i.imgur.com/Q2RNLXY.png)

**Comment by mattbruv** [comment]
Had this happen again, basically it seems to happen on the pyramid level after I've bought the second bomb, and spam press A to rapidly drop bombs. The second time it also is stuck in the same loop

**Comment by 1r3n33** [comment]
Hello @mattbruv,
Thank you for reporting this issue.
I have been able to reproduce the softlock but got stuck at a different address.
However, like in your case, it seems to be always happening in `spcProcessStream` (snesmod library) while syncing with the SPC:
```
-:	cpx	REG_APUIO0		; wait for spc
	bne	-			;
```
I'll keep you posted on further investigation.

**Comment by mattbruv** [comment]
Thanks. I was trying to debug it earlier, but couldn't get the ROM to build correctly after fighting the makefile for a while. I'm interested to know what the problem/fix is.

**Comment by 1r3n33** [comment]
@mattbruv
> ...couldn't get the ROM to build correctly after fighting the makefile for a while.

Would you please elaborate on what was preventing you from building?
Some requirements can be found in [README](https://github.com/1r3n33/bomberworld/blob/master/README.md).

**Comment by mattbruv** [comment]
Yeah, I already had PvSneslib installed, I was able to get the ROM to build but the makefile would still error after linking for some reason. When playing it there was something wrong with the sprites. When I dropped bombs the sprites weren't there or looked corrupted. It's like it was able to build, but not perfectly. I kind of gave up on trying to build it after fighting with it for a while

**Comment by 1r3n33** [comment]
I see. The game requires a specific branch of pvsneslib: `git clone -b local-dev-env git@github.com:1r3n33/pvsneslib.git`
Unfortunately, the latest official version of pvsneslib produces the kind of glitches you described.

**Comment by mattbruv** [comment]
Using an old version could be the reason for the softlock too, as when I tried to inspect the routine it hangs on in the latest branch, it appeared that the ASM is different
